src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -70,8 +71,46 @@ def internal_validate_lock_parameters(resource_group_name, resource_provider_nam
 
 
 def validate_lock_parameters(namespace):
-    internal_validate_lock_parameters(getattr(namespace, 'resource_group_name', None),
+    internal_validate_lock_parameters(getattr(namespace, 'resource_group', None),
                                       getattr(namespace, 'resource_provider_namespace', None),
                                       getattr(namespace, 'parent_resource_path', None),
                                       getattr(namespace, 'resource_type', None),
                                       getattr(namespace, 'resource_name', None))
+
+
+def validate_subscription_lock(namespace):
+    if getattr(namespace, 'ids', None):
+        for lock_id in getattr(namespace, 'ids'):
+            if not _parse_lock_id(lock_id).get('resource_group_name'):
```

Comment:
Not very familiar with locks, but if having `resource_group_name` is an error, which i believe so, then you should remove the `not`, or you should check `subscription` is the only available value which might be simpler, because I assume having resource related fields would be an error case as well.

Reply:
Also I would suggest error our on any locks applying on tenant-level accounts(those with no actual subscription). Optionally, but good to have